src/http/client/builder.rs

```diff
@@ -415,115 +402,48 @@ impl<T> EasyHttpWebClientBuilder<T, ProxyStage> {
             _phantom: PhantomData,
         }
     }
 
-    #[cfg(any(feature = "rustls", feature = "boring"))]
-    /// Add a http request inspector that will run just after the inner http connector
-    /// has connected but before the http handshake happens
-    ///
-    /// This will also add the [`HttpsAlpnModifier`] request inspector as that one is
-    /// crucial to make tls alpn work and set the correct [`TargetHttpVersion`]
-    ///
-    /// And a [`HttpVersionAdapter`] that will adapt the request version to the configured
-    /// [`TargetHttpVersion`]
-    ///
-    /// If you don't want any of these inspector you can use [`Self::with_advanced_jit_req_inspector`]
-    /// to configure your own request inspectors without any defaults
-    ///
-    /// [`TargetHttpVersion`]: crate::http::conn::TargetHttpVersion
-    pub fn with_jit_req_inspector<I>(
+    /// Add a custom http connector that will be run just after tls
+    pub fn with_custom_http_connector<L>(
         self,
-        http_req_inspector: I,
-    ) -> EasyHttpWebClientBuilder<
-        HttpConnector<T, (HttpsAlpnModifier, HttpVersionAdapter, I), I2>,
-        HttpStage,
-    > {
-        EasyHttpWebClientBuilder {
-            connector: self.connector.with_jit_req_inspector((
-                HttpsAlpnModifier::default(),
-                HttpVersionAdapter::default(),
-                http_req_inspector,
-            )),
-            _phantom: PhantomData,
-        }
-    }
+        connector_layer: L,
+    ) -> EasyHttpWebClientBuilder<L::Service, HttpStage>
+    where
+        L: Layer<T>,
+    {
+        let connector = connector_layer.into_layer(self.connector);
 
-    #[cfg(not(any(feature = "rustls", feature = "boring")))]
-    /// Add a http request inspector that will run just after the inner http connector
-    /// has finished but before the http handshake
-    ///
-    /// This will also add the [`HttpVersionAdapter`] that will adapt the request version to
-    /// the configured [`TargetHttpVersion`]
-    ///
-    /// If you don't want any of these inspector you can use [`Self::with_advanced_jit_req_inspector`]
-    /// to configure your own request inspectors without any defaults
-    ///
-    /// [`TargetHttpVersion`]: crate::http::conn::TargetHttpVersion
-    pub fn with_jit_req_inspector<I>(
-        self,
-        http_req_inspector: I,
-    ) -> EasyHttpWebClientBuilder<HttpConnector<T, (HttpVersionAdapter, I), I2>, HttpStage> {
         EasyHttpWebClientBuilder {
-            connector: self
-                .connector
-                .with_jit_req_inspector((HttpVersionAdapter::default(), http_req_inspector)),
+            connector,
             _phantom: PhantomData,
         }
     }
+}
 
+impl<T, I> EasyHttpWebClientBuilder<HttpConnector<T, I>, HttpStage> {
     /// Add a http request inspector that will run just before doing the actual http request
-    pub fn with_svc_req_inspector<I>(
+    pub fn with_svc_req_inspector<X>(
```

Comment:
Err.... Postbus X? To young I guess.

Anyway... I would call this `U`, in spirit of the alphabet etc.